beets/util/__init__.py

```diff
@@ -500,6 +501,31 @@ def link(path, dest, replace=False):
                               traceback.format_exc())
 
 
+def hardlink(path, dest, replace=False):
+    """Create a hard link from path to `dest`. Raises an OSError if
+    `dest` already exists, unless `replace` is True. Does nothing if
+    `path` == `dest`."""
+    if (samefile(path, dest)):
+        return
+
+    path = syspath(path)
+    dest = syspath(dest)
```

Comment:
It might seem like a shortcut, but doing this once and overwriting the variables will cause problems: namely, the exceptions below use `path` and `dest`, which will now be system paths. This will produce strange results on Windows, where a ridiculous prefix needs to be added for `syspath` results.